ucsm_apis/admin/callhome.py

```diff
@@ -0,0 +1,803 @@
+# Copyright 2017 Cisco Systems, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module performs the operation related to callhome.
+"""
+from ucsmsdk.ucsexception import UcsOperationError
+
+_base_dn = "call-home"
+
+
+def call_home_enable(handle, alert_throttling_admin_state="on",
+                     policy_owner="local",name=None, descr=None, **kwargs):
+    """
+    Enables call home alert.
+
+    NOTE:
+	If enabling call home for the first time then first update contact and
+	smtp server.
+	- contact information using 'call_home_contact_update'
+		call_home_contact_update(handle,
+							     contact="ciscoucs",
+                                 phone="+911234567890",
+                                 email="ciscoucs@cisco.com",
+                                 addr="cisco",
+							     reply_to="ciscoucs@cisco.com"
+                                 )
+    - smtp server using 'call_home_smtp_update'.
+		call_home_smtp_update(handle, host="1.1.1.1")
+
+
+    Args:
+        handle (UcsHandle)
+        alert_throttling_admin_state (string): "on" or "off"
+        name (string): name
+        descr (string): description
+        **kwargs: Any additional key-value pair of managed object(MO)'s
+                  property and value, which are not part of regular args.
+                  This should be used for future version compatibility.
+    Returns:
+        CallhomeEp : ManagedObject
+
+    Raises:
+        UcsOperationError: If CallhomeEp is not present
+
+    Example:
+        call_home_enable(handle, alert_throttling_admin_state="on")
+    """
+    mo = handle.query_dn(_base_dn)
+    if not mo:
+        raise UcsOperationError("call_home_state_enable",
+                                 "Call home not available.")
+
+    mo.admin_state = "on"
```

Comment:
better to be consistent and write this as `mo.set_prop_multiple(**args)` with vars defined in args